pootle/apps/pootle_store/utils.py

```diff
@@ -176,14 +176,14 @@ def add(self, unit, translation, user=None):
         if isinstance(user, User):
             user = user.id
         user = user or User.objects.get_system_user().id
-        pending = SuggestionState.objects.get(name="pending")
         try:
             suggestion = Suggestion.objects.pending().get(
                 unit=unit,
                 user_id=user,
                 target_f=translation)
             return (suggestion, False)
         except Suggestion.DoesNotExist:
+            pending = SuggestionState.objects.get(name="pending")
```

Comment:
You can just get the `id` using `values_list('id', flat=True)` but not sure if that will optimize it further.